torch/csrc/autograd/custom_function.cpp

```diff
@@ -133,13 +139,14 @@ variable_list _wrap_outputs(const variable_list &input_vars,
 
     outputs_impl.insert(out_tensor_impl);
     outputs.emplace_back(var);
+    num_tensors++;
   }
 
   // If multiple differentiable outputs are returned, we do not allow views to be modified inplace
   // See NOTE [ View + Inplace detection ] for more details
   if (num_diff_outputs > 1) {
     for (auto& var: outputs) {
-      if (var.is_view()) {
+      if (var.defined() && var.is_view()) {
```

Comment:
Why do you need to check if var is defined here? You bail out at the beginning of the loop in the other case no?